### PR TITLE
Added a warning if `flutter.groovy` uses a `.flutter-plugins` file.

### PR DESCRIPTION
**Pull request by matanlurey**
Work towards https://github.com/flutter/flutter/issues/48918.

This file was soft deprecated in _2020_, but the code was never removed. This warning message will serve as a warning and we'll rip out support for `flutter-plugins` after the _next_ stable release (i.e. after the mid-November branch cut).

### CODE REVIEW

**Review by reidbaker — APPROVED** [review]

File: packages/flutter_tools/gradle/src/main/groovy/flutter.groovy [hunk 1]
```diff
@@ -688,6 +688,10 @@ class FlutterPlugin implements Plugin<Project> {
         } catch (FileNotFoundException ignored) {
             throw new GradleException("settings.gradle/settings.gradle.kts does not exist: ${settingsGradleFile(project).absolutePath}")
         }
+        // TODO(matanlurey): https://github.com/flutter/flutter/issues/48918.
+        project.logger.quiet("Warning: This project is still reading the deprecated '.flutter-plugins. file.")
+        project.logger.quiet("In an upcoming stable release support for this file will be completely removed and your build will fail.")
+        project.logger.quiet("For an example of updating Android build scripts, see https://docs.flutter.dev/release/breaking-changes/flutter-gradle-plugin-apply.")
```
Comment (reidbaker) [inline packages/flutter_tools/gradle/src/main/groovy/flutter.groovy:694]: Is this the right link? I dont see .flutter-plugins anywhere on this page (which I would expect)

Comment (matanlurey) [reply packages/flutter_tools/gradle/src/main/groovy/flutter.groovy:694]: There is no canonical link for "keep your gradle files up to date", is there?

Comment (reidbaker) [reply packages/flutter_tools/gradle/src/main/groovy/flutter.groovy:694]: but this isnt "keep your gradle files up to date" i dont think

.flutter-plugins was the way and we changed the way so we should have migration explanation someplace.

Comment (matanlurey) [reply packages/flutter_tools/gradle/src/main/groovy/flutter.groovy:694]: Ok I can write something.

Comment (reidbaker) [reply packages/flutter_tools/gradle/src/main/groovy/flutter.groovy:694]: I searched for the documentation and cant find any https://github.com/search?q=org%3Aflutter+.flutter-plugins&type=code

Comment (reidbaker) [reply packages/flutter_tools/gradle/src/main/groovy/flutter.groovy:694]: https://github.com/flutter/flutter/pull/45379 possibly related pr

**Review by reidbaker — APPROVED** [review]

**Comment by matanlurey** [comment]
Reason for revert: We have `--fatal-warnings` on postsubmit and this breaks the tree.

**Pull request merged**